HPXMLtoOpenStudio/resources/hpxml.rb

```diff
@@ -2192,6 +2258,123 @@ def from_doc(climate_zone_iecc)
     end
   end
 
+  class Zones < BaseArrayElement
+    def add(**kwargs)
+      self << Zone.new(@parent_object, **kwargs)
+    end
+
+    def from_doc(building)
+      return if building.nil?
+
+      XMLHelper.get_elements(building, 'BuildingDetails/Zones/Zone').each do |zone|
+        self << Zone.new(@parent_object, zone)
+      end
+    end
+  end
+
+  class Zone < BaseElement
+    def initialize(hpxml_bldg, *args)
+      @spaces = Spaces.new(hpxml_bldg)
```

Comment:
@shorowit  I had a hard time tracking the history of this, but the Space initialization here uses hpxml_bldg as parent object, while in from_doc, it uses self (which is zone) as parent object, this looks a bit confusing to me. Is this intended?

Reply:
Can you compare to other similar classes that have been in hpxml.rb for a long time? I would use what they've been using.

Reply:
Yep I found Ducts attribute at HVACDistribution is quite similar and I changed it to be consistent with that, where we only call from_doc instead of reinitializing.